txn.go

```diff
@@ -276,14 +324,26 @@ func (tr *transactionRunner) Run(transactions TransactionSource) error {
 			msg := err.Error()
 			retryErr := strings.HasSuffix(msg, "unexpected message") ||
 				strings.HasSuffix(msg, "i/o timeout")
-			if !retryErr || i == (nrRetries-1) {
+			if !retryErr || i == (tr.nrRetries-1) {
 				return err
 			}
 		}
 	}
 	return ErrExcessiveContention
 }
 
+func (tr *transactionRunner) pause(attempt int) {
+	// Backoff a little longer each failed attempt and throw in
+	// a bit of fuzz for good measure.
+	fuzz := rand.Float32()
+	dur := time.Microsecond * time.Duration(
+		// Increase the backoff time for each attempt.
+		int(tr.retryBackoffMillis*1000)*attempt+
+			// Include a random amount of time as well.
+			int(tr.retryFuzzMillis*fuzz*1000))
```

Comment:
our fuzz is also usually +/- not just strictly longer than the base retry.
I'm not terribly concerned about 2ms delay, but it is good to have consistency around the patterns.

A lot of this feels like it is clearer if you started with time.Duration rather than Micro* (Millis*1000.)